profiling/src/exporter/mod.rs

```diff
@@ -162,12 +162,10 @@ impl ProfileExporter {
     /// "RFC: Pprof System Info Support".
     pub fn build(
         &self,
-        start: DateTime<Utc>,
-        end: DateTime<Utc>,
+        profile: EncodedProfile,
         files_to_compress_and_export: &[File],
         files_to_export_unmodified: &[File],
         additional_tags: Option<&Vec<Tag>>,
-        endpoint_counts: Option<&ProfiledEndpointsStats>,
         internal_metadata: Option<serde_json::Value>,
         info: Option<serde_json::Value>,
     ) -> anyhow::Result<Request> {
```

Comment:
Consider perhaps making `profile` optional? In particular, that enables experimentation around e.g. if I don't want to send a profile, or want to post-process the profile before sending it. (It's not a very strong use-case, but making it optional seems quite simple, and it'll be so annoying to otherwise have to create "dummy" profiles just to get around this if it's not)

Reply:
[`3786ad1` (#926)](https://github.com/DataDog/libdatadog/pull/926/commits/3786ad1a627df505efcc1274c0a99571eccf6699)

It makes the code a bit uglier, since we have to handle option some places. LMK if you think its worth it

Reply:
Rolled back, as per other discussion